File: 0134-gas-station/0134-gas-station.py
class Solution(object):
    def canCompleteCircuit(self, gas, cost):
        gs = 0
        cs = 0
        start = 0
        tot = 0

        for it in gas:
            gs += it

        for it in cost:
            cs += it

        if gs < cs:
            return -1

        for i in range(len(gas)):
            tot += gas[i] - cost[i]

            if tot < 0:
                start = i + 1
                tot = 0

        return start

File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int gs=0,cs=0;

        for(auto it:gas){
            gs+=it;
        }
        for(auto it:cost){
            cs+=it;
        }

        if(gs<cs)return -1;

        int start=0;
        int css=0;

        for(int i=0;i<gas.size();i++){
            css+=gas[i]-cost[i];
            if(css<0){
                start=i+1;
                css=0;
            }
        }
        return start;
    }
};